roxie/ccd/ccdserver.cpp

```diff
@@ -553,53 +557,51 @@ class CRoxieServerActivityFactoryBase : public CActivityFactory, implements IRox
     {
         dbgassertex(!_idx);
         if (likely(_processed))
-            processed += _processed;
+            myedgestats.addStatistic(StNumRowsProcessed, _processed);
     }
 
     virtual void noteStarted() const
     {
-        started++;
+        myedgestats.addStatistic(StNumStarts, 1);
+        myedgestats.addStatistic(StNumStops, 1);    // Makes things prettier - we don't actually track stops at the factory level
+        myedgestats.setStatistic(StNumSlaves, 1);   // Ditto - meaningless for Roxie but affects the display
     }
 
     virtual void noteStarted(unsigned idx) const
     {
         throwUnexpected(); // should be implemented/required by multiOutput cases only
     }
 
-    virtual void getEdgeProgressInfo(unsigned output, IPropertyTree &edge) const
-    {
-        if (output == 0)
-        {
-            putStatsValue(&edge, "NumRowsProcessed", "sum", processed);
-            auto _started = started.load();
-            if (_started)
-                putStatsValue(&edge, "NumStarts", "sum", _started);
-        }
-        else
-            IERRLOG("unexpected call to getEdgeProcessInfo for output %d in activity %d", output, queryId());
-    }
-
-    virtual void getNodeProgressInfo(IPropertyTree &node) const
-    {
-        CActivityFactory::getNodeProgressInfo(node);
-        auto _started = started.load();
-        if (_started)
-            putStatsValue(&node, "_roxieStarted", "sum", _started);
-    }
-
     virtual void resetNodeProgressInfo()
     {
         CActivityFactory::resetNodeProgressInfo();
-        processed = 0;
-        started = 0;
     }
     virtual void getActivityMetrics(StringBuffer &reply) const
```

Comment:
cleanup: remove the override, since it just calls the base?

Reply:
It's needed to implement the interface